### PR TITLE
New front for gradio chapters

### PR DESCRIPTION
**Pull request by mishig25**
Use this branch `gradio_newfront` as your base branch when you are making changes to chapter 9 until we deploy chapter 9

You can test your changes locally with:
```
pip install git+https://github.com/huggingface/doc-builder

doc-builder preview course ~/Desktop/course/chapters/en --not_python_module
```
Note: make sure to change `~/Desktop/course/chapters/en` based on your filesystem

The command above should open http://localhost:3000/ automaatically. You'll see 404 on http://localhost:3000/, ignore it and just click whatever section you are editing on the left navbar.

Also, hot reloading is enabled by default, you should be able to see your changes live

Screenshot 2022-03-23 at 16 29 24

## TODO

- [x] Add sections on Gradio blocks
- [x] Check if any old sections need updating for Gradio V3
- [ ] Generate notebooks for Colab
- [ ] Add note about Block Party event at start and end of chapter

### CODE REVIEW

**Comment by lewtun** [comment]
FYI I've integrated the changes on `main` so we can automatically build the docs in the CI - let's see if it works 🤞

**Comment by lewtun** [comment]
OK the docs are failing to build due to a strange error (does not happen on `main`):

```
TypeError: Cannot read properties of undefined (reading 'length')
    at each (file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/chunks/index-2bd0fec1.js:83:29)
    at file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/chunks/Question-8d0b1575.js:58:24
    at Object.$$render (file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/chunks/index-2bd0fec1.js:112:18)
    at file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/entries/pages/chapter9/7.mdx.js:59:44
    at Object.$$render (file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/chunks/index-2bd0fec1.js:112:18)
    at Object.default (file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/index.js:57:92)
    at file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/entries/pages/__layout.svelte.js:15:68
    at Object.$$render (file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/chunks/index-2bd0fec1.js:112:18)
    at file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/index.js:51:97
    at $$render (file:///tmp/tmplofvkojh/kit/.svelte-kit/output/server/chunks/index-2bd0fec1.js:112:18)
> 500 /docs/course/pr_34/en/chapter9/7
    at file:///tmp/tmplofvkojh/kit/node_modules/@sveltejs/kit/dist/chunks/index5.js:427:11
    at save (file:///tmp/tmplofvkojh/kit/node_modules/@sveltejs/kit/dist/chunks/index5.js:659:4)
    at visit (file:///tmp/tmplofvkojh/kit/node_modules/@sveltejs/kit/dist/chunks/index5.js:[55](https://github.com/huggingface/course/runs/5833335009?check_suite_focus=true#step:10:55)2:3)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

Installing node dependencies
Building HTML files. This will take a while :-)
Traceback (most recent call last):
  File "/usr/local/bin/doc-builder", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/doc_builder/commands/doc_builder_cli.py", line 43, in main
    args.func(args)
  File "/usr/local/lib/python3.8/site-packages/doc_builder/commands/build.py", line 143, in build_command
    subprocess.run(
  File "/usr/local/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['npm', 'run', 'build']' returned non-zero exit status 1.
```

Maybe @mishig25 can help us understand what's causing this when he gets back from vacation 😎

**Review by lewtun — APPROVED** [review]
This looks good to me! Should we merge now or wait until we've updated the content with `gradio` v3?

File: chapters/en/_toctree.yml [hunk 1]
```diff
@@ -167,6 +167,27 @@
     title: End-of-chapter quiz
     quiz: 8
 
+- local: chapter9
+  title: 9. Building and sharing demos
+  new: true
```
Comment (lewtun) [inline chapters/en/_toctree.yml:172]: FYI this field will eventually render a "New" icon on the frontend - see internal Slack thread: https://huggingface.slack.com/archives/C01SJL5C5LK/p1651736595622049

**Comment by mishig25** [comment]
@lewtun please let me know if there's anything I need to do for the new gradio chapters

File: chapters/en/_toctree.yml (hunk 1 above)
Comment (lewtun) [reply chapters/en/_toctree.yml:172]: @mishig25 should I expect to see the "new" icon already in this PR or is this something that will become visible once I re-build the content on the `release` branch (for prod)?

Screen Shot 2022-05-10 at 11 53 40

File: chapters/en/chapter9/6.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+image = gr.inputs.Image(shape=(224, 224))
+label = gr.outputs.Label(num_top_classes=3)
+
+title = "Gradio Image Classifiction + Interpretation Example"
+gr.Interface(
+    fn=classify_image, inputs=image, outputs=label, interpretation="default", title=title
+).launch(auth=("admin", "pass1234"))
+```
+
+<iframe src="https://hf.space/gradioiframe/dawood/Auth-Image-Classification/+" frameBorder="0" height="570" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```
Comment (lewtun) [inline chapters/en/chapter9/6.mdx:135]:
```suggestion
<iframe src="https://hf.space/gradioiframe/dawood/Auth-Image-Classification/+" frameBorder="0" height="570" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>

This wraps up our deep dive into the `Interface` class of Gradio. As we've seen, this class makes it simple to create machine learning demos in a few lines of Python code. However, sometimes you'll want to customise your demo by changing the layout or chaining multiple prediction functions together. Wouldn't it be nice if we could somehow split the `Interface` into customizable "blocks"? Fortunately, there is! That's the topic of the final section.
```

**Review by lewtun — APPROVED** [review]
I've done a final pass through the whole chapter and reading it again was a real pleasure - excellent job @dawoodkhan82 and @abidlabs 😍 !

I've added a bunch of nits to streamline the text and formatting with the other course chapters, but it's mostly minor stuff that should be easy to include :)

One last thing: @osanseviero suggested mentioning the Blocks event in `chapter9/1` and `chapter9/7`. How about we include a small description of the event at the end of each of these sections?

Once we merge this, I'll then generate all the Colab notebooks so people can easily play with the demos :)

**Comment by abidlabs** [comment]
LGTM, love the cute emojis sprinkled in the chapter 😄

File: chapters/en/_toctree.yml (hunk 1 above)
Comment (abidlabs) [reply chapters/en/_toctree.yml:172]: Sounds good

**Review by abidlabs — APPROVED** [review]

**Pull request merged**